android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +80,102 @@ public void init(String siteKey, String touchpointUri, String apiKey,
   }
 
   @ReactMethod
-  public void sendInteraction(String interaction, ReadableMap propertiesMap, final Promise promise) {
+  public void sendInteraction(String interactionPath, ReadableMap propertiesMap, final Promise promise) {
     HashMap<String, String> properties = getPropertiesFromReadableMap(propertiesMap);
     final OneRequest sendInteractionRequest = new OneRequest.Builder()
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
+      .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
       .properties(properties)
       .build();
-    final OneCall sendInteractionCall = One.sendInteraction(sendInteractionRequest);
-
-    if (promise == null) {
-      sendInteractionCall.enqueue(null);
-    } else {
-      sendInteractionCall.enqueue(new OneCallback() {
-        @Override
-        public void onSuccess(@NotNull OneResponse oneResponse) {
-          notifyResult(promise,
-            "{\"captures: " + oneResponse.getCaptures().toString() + ", " +
-                  "\"interactionPath\": \"" + oneResponse.getInteractionPath().getValue().toString() + "\", " +
-                  "\"optimizations: " + oneResponse.getOptimizations().toString() + ", " +
-                  "\"statusCode\": " + oneResponse.getHttpStatusCode() + ", " +
-                  "\"tid\": \"" + oneResponse.getTid() + "\", " +
-                  "\"trackers\": " + oneResponse.getTrackers().toString() + "}"
-          );
-        }
-
-        @Override
-        public void onError(@NotNull OneSDKError oneSDKError) {
-          notifyProblem(promise, "" + oneSDKError.getSystemCode(), oneSDKError.getErrorMessage());
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendInteraction(true, sendInteractionRequest).join();
+        } else {
+          response = One.sendInteractionLegacySupport(true, sendInteractionRequest).join();
         }
-
-        @Override
-        public void onFailure(@NotNull OneAPIError oneAPIError) {
-          notifyProblem(promise, "" + oneAPIError.getHttpStatusCode(), oneAPIError.getErrorMessage());
-        }
-      });
-    }
+        One.processResponse(response);
```

Comment:
Sorry I am new to this project but this seems to me like we have a problem with the react api.  When calling the native api (ios or android) the only thing the api automatically does is keep track of the TID.  It does not process anything (optimizations or captures).  It only proceeds to do that when the `processResponse` api is called.  

Calling `One.processResponse` here does not follow that flow.  The caller will get the result after the native SDK has started to do its thing.

I see however we don't have a "process response" api though in react.  Looking at the old cold the process was never called so this is new.  Did you mean to add it? I understand we don't have automatic capabilities right now in react so this may be a nothingburger but I feel we should not call process here without a plan.  If its not needed then please remove.

Reply:
Removed